### DISCUSSION TITLE
Network Returns VPN as WiFi - Bug or Working as intended

### DISCUSSION THREAD

**Discussion by Wahsner** [ionic-team/capacitor#5277] [General]
Hello,

Im at a loss here and I'm not sure if this is a bug or what!
I need to check if the device is online / offline so I installed the capacitorjs network package.
However, it appears on iOS (unable to test/confirm on Android) that a VPN connection is being/returning as Connected = True and of Type WiFi

This is leading to issues because there isnt a internet connection for the VPN. I need to ensure the JWT isnt refreshed while offline but due to this it is attempting to refresh and unable too leading to user being logged out.

**Comment by kensodemann** [comment]
The behaviour you are seeing sounds correct to me.

The Network plugin will ask iOS for its network status, and iOS will give it. This is a completely separate question as to whether or not the Internet is accessible via that network.

For example, I can be connected to my home WiFi (for example) and my ISP could be down. I am still connected to my home WiFi, however, so iOS will (correctly) report that I am connected via WiFi, because I am. Whether or not the Internet is available via that network connection is a completely different question.

For that matter, the Internet could be there and 100% accessible, but your server that you will try to refresh your JWT with could be unavailable.

**Reply by Wahsner** [reply]
Hi @kensodemann 

Thank you for your response.

After some further testing and digging I can confirm that, in relation to @capacitor/network, If there is an active VPN Client on the users device with kill switches and “always on” features. (like Mullvad) the following three issues occur;
  1) The connection is incorrectly and/or unexpectedly being reported as 'True'
  2) The connection type is incorrectly and/or unexpectedly being reported as 'WiFi
  3) The networkStatusChange listener isnt triggered 
The above scenario also presumes the device is disconnected from all network types either manually via airplane mode and/or disabling WiFi / cellular or naturally via being in an area where there is no network avaliability.

Therefore, this isnt a bug. But this is behaviour which I think should be known/added somewhere for other people to be aware of.
